src/iter/italgos.h

```diff
@@ -112,6 +112,15 @@ double power(unsigned int maxiter,
 	struct iter_op_s op,
 	float* u);
 	   
+void pridu(unsigned int maxiter, float epsilon, float tau, float sigma, float theta, float decay,
+	long N, long M,
+	const struct vec_iter_s* vops,
+	struct iter_op_s op_forw,
+	struct iter_op_s op_adj,
+	struct iter_op_p_s thresh1,
+	struct iter_op_p_s thresh2,
+	float* x,
+	struct iter_monitor_s* monitor);
 
```

Comment:
There are many primal dual algorithms. Maybe call it chambolle_pock or something?

Reply:
Sure